src/tactic/interactive.lean

```diff
@@ -352,7 +352,10 @@ do    tgt ← target,
       gs ← with_enable_tags (
         mzip_with (λ (n : name × name) v, do
            set_goals [v],
-           try (interactive.unfold (provided.map $ λ ⟨s,f⟩, f.update_prefix s) (loc.ns [none])),
+           b ← target >>= is_prop,
+           if b then
+             try (interactive.dunfold (provided.map $ λ ⟨s,f⟩, f.update_prefix s) (loc.ns [none]))
+           else return (),
```

Comment:
I tried this instead of testing for `is_prop` as suggested by @gebner, and the result contains `id`. What am I doing wrong?
```suggestion
           try (do
             tgt ← target,
             interactive.dunfold (provided.map $ λ ⟨s,f⟩, f.update_prefix s) (loc.ns [none]),
             tgt' ← target,
             guard (tgt ≠ tgt')),
```

Reply:
This is interesting to debug.  If you just insert `trace (tgt, tgt')` then you are still completely clueless.  But if you insert `trace tgt.to_raw_fmt, trace tgt'.to_raw_fmt` then you see why `tgt ≠ tgt'`:
```
(mvar _mlocal._fresh.192.805 _mlocal._fresh.192.805 (const 2 []))
(pi x default (local_const 0._fresh.192.45 α (const 1 [])) (app (local_const 0._fresh.192.49 β (const 1 [])) (var 0)))
```

The fix is obviously to write `tgt ← target >>= instantiate_mvars`.

Reply:
How something involving `instantiate_mvars` can be obvious? Should I do it for `tgt`, `tgt'`, or both?

Reply:
It is obvious because the output of `trace tgt` is `Π (x : α), β x` while the output of `trace tgt.to_raw_fmt` is a metavariable.  Hence `tgt` contains uninstantiated metavariables, and we need to call `instantiate_mvars` to fix this.

I think it should be enough to do it for `tgt`.  :crossed_fingers:

Reply:
Maybe I should clarify one more point: when you evaluate `tgt = tgt'`, then this of course checks the equality of the expressions without regard to the metavariable assignment.  And a metavariable is not equal to a pi.